notifier/notifier.go

```diff
@@ -553,7 +564,7 @@ type alertmanagerSet struct {
 	logger     log.Logger
 }
 
-func newAlertmanagerSet(cfg *config.AlertmanagerConfig, logger log.Logger) (*alertmanagerSet, error) {
+func newAlertmanagerSet(cfg Config, logger log.Logger) (*alertmanagerSet, error) {
```

Comment:
You're passing around full structs rather than pointers, just pointers would be more efficient.